redash/query_runner/trino.py

```diff
@@ -83,26 +83,48 @@ def type(cls):
         return "trino"
 
     def get_schema(self, get_stats=False):
-        query = """
-            SELECT table_schema, table_name, column_name, data_type
-            FROM information_schema.columns
-            WHERE table_schema NOT IN ('pg_catalog', 'information_schema')
-        """
-        results, error = self.run_query(query, None)
+        if self.configuration.get("catalog"):
+            catalog_prefixes = [""]
+        else:
+            query = """
+                SHOW CATALOGS
+            """
+            results, error = self.run_query(query, None)
+
+            if error is not None:
+                self._handle_run_query_error(error)
+
+            results = json_loads(results)
 
-        if error is not None:
-            self._handle_run_query_error(error)
+            catalog_prefixes = []
+            for row in results["rows"]:
+                catalog = row["Catalog"]
+                if "." in catalog:
+                    catalog = f'"{catalog}"'
+                catalog_prefixes.append(f"{catalog}.")
```

Comment:
The catalog prefix having a `.` suffix looks weird to me, but I don't know how to make the pattern better. Is there a way for the provided catalog to be picked up instead of setting the prefixes to `[""]`?

Reply:
Fixed in c682e01